e2e/third-party-donation.spec.ts

```diff
@@ -0,0 +1,116 @@
+    ).toBeVisible();
+
+    await expect(page.getByRole('tab', { name: '$5' })).toBeVisible();
+    await expect(page.getByRole('tab', { name: '$10' })).toBeVisible();
+    await expect(page.getByRole('tab', { name: '$20' })).toBeVisible();
+    await expect(page.getByRole('tab', { name: '$40' })).toBeVisible();
+  });
+
+  test('The donation widget should let you know what you"ll support per donation value', async ({
+    page
+  }) => {
+    await page.getByRole('tab', { name: '$5' }).click();
+    await expect(page.getByTestId('donation-tier-selector')).toContainText(
+      'Confirm your donation of $5 / month:'
+    );
+    await expect(page.locator('.tab-content').nth(0)).toContainText(
+      'Your $5 donation will provide 250 hours of learning to people around the world each month.'
+    );
+
+    await page.getByRole('tab', { name: '$10' }).click();
+    await expect(page.getByTestId('donation-tier-selector')).toContainText(
+      'Confirm your donation of $10 / month:'
+    );
+    await expect(page.locator('.tab-content').nth(1)).toContainText(
+      'Your $10 donation will provide 500 hours of learning to people around the world each month.'
+    );
+
+    await page.getByRole('tab', { name: '$20' }).click();
+    await expect(page.getByTestId('donation-tier-selector')).toContainText(
+      'Confirm your donation of $20 / month:'
+    );
+    await expect(page.locator('.tab-content').nth(2)).toContainText(
+      'Your $20 donation will provide 1,000 hours of learning to people around the world each month.'
+    );
+
+    await page.getByRole('tab', { name: '$40' }).click();
+    await expect(page.getByTestId('donation-tier-selector')).toContainText(
+      'Confirm your donation of $40 / month:'
+    );
+    await expect(page.locator('.tab-content').nth(3)).toContainText(
+      'Your $40 donation will provide 2,000 hours of learning to people around the world each month.'
+    );
+  });
+
+  test('When the donation button is clicked it should tell you that it is secure', async ({
+    page
+  }) => {
+    await page.getByRole('button', { name: 'Donate', exact: true }).click();
+
+    await expect(page.getByText('Secure donation')).toBeVisible();
+  });
+
+  // We do NOT want to test the implementation of the third-party services
+  // We only want to test that the buttons are there (our buttons ^) and that they are visible
+  // and that the donation widget works as expected if we do decide on testing the implementation
+  // of the third-party services this will be a good starting point
+
+  // test('It should have a Paypal and Patreon button', async ({ page }) => {
+  //   await page.getByRole('button', { name: 'Donate', exact: true }).click();
+  //   await expect(
+  //     page.locator('[data-paypal-smart-button-version="5.0.440"]')
+  //   ).toBeVisible();
+  //   await expect(
+  //     page.locator('[data-patreon-widget-type="become-patron-button"]')
+  //   ).toBeVisible();
+  // });
```

Comment:
Can this part be uncommented and included in tests?

Can we not test the presence of the PayPal and Patreon buttons without clicking them for the actual API call to those services? 

That is, the user story for the test will be: As I user I would like to select an option for the vendor (but not proceed further by clicking on it)

Does that makes sense? If yes, then can you do the same for the Stripe stuff below as well?

Reply:
The buttons are not present without the API Keys

Reply:
Meaning if we want to test this we would include the secrets in the Playwright workflow or a separate workflow that only runs on the staging and/or production branch.

Reply:
Ahmad says, we only need client-side keys for that. Client-side keys are not sensitive and are part of our client app - shipped to browsers.